imagehash.py

```diff
@@ -390,10 +390,21 @@ def __eq__(self, other):
 			return False
 		return self.matches(other)
 
+	def __ne__(self, other):
+		return not self.matches(other)
+
 	def __sub__(self, other, hamming_cutoff=None, bit_error_rate=None):
 		matches, sum_distance = self.hash_diff(other, hamming_cutoff, bit_error_rate)
+		max_difference = len(self.segment_hashes)
+		if matches == 0:
+			return max_difference
 		max_distance = matches * len(self.segment_hashes[0])
-		return matches + (1 - (sum_distance / max_distance))
+		tie_breaker = 1 - (float(sum_distance) / max_distance)
```

Comment:
Presumably sum_distance can take values between 0 and max_distance, and therefore tie_breaker is between 0 and 1.
When matches == max_difference - 1, and sum_distance == 0, `__sub__` returns zero. Is that good?

Reply:
That.. is not good.
If an image matches all segment hashes, so matches = max_difference, and sum_distance == 0, __sub__ returns -1. Very not right.

That `tie_breaker = 1 - ...` should be `tie_breaker = 0 - ...`

Then tie_breaker is between -1 and 0. That seems fine
I'll change that out.